nextroute/schema/input.go

```diff
@@ -7,129 +7,217 @@ import (
+	MinStopsPenalty *float64 `json:"min_stops_penalty,omitempty"`
+	// MaxStops maximum number of stops that the vehicle can visit.
+	MaxStops *int `json:"max_stops,omitempty"`
+	// MaxDistance maximum distance in meters that the vehicle can travel.
+	MaxDistance *int `json:"max_distance,omitempty"`
+	// MaxDuration maximum duration in seconds that the vehicle can travel.
+	MaxDuration *int `json:"max_duration,omitempty"`
+	// MaxWait maximum aggregated waiting time that the vehicle can wait across route stops.
+	MaxWait *int `json:"max_wait,omitempty"`
+	// CompatibilityAttributes attributes that the vehicle is compatible with.
+	CompatibilityAttributes *[]string `json:"compatibility_attributes,omitempty"`
+	// ActivationPenalty penalty of using the vehicle.
+	ActivationPenalty *int `json:"activation_penalty,omitempty"`
+	// AlternateStops a set of alternate stops for which only one should be serviced.
+	AlternateStops *[]string `json:"alternate_stops,omitempty"`
 }
 
 // StopDefaults contains default values for stops.
 type StopDefaults struct {
-	UnplannedPenalty        *int       `json:"unplanned_penalty,omitempty"`
-	Quantity                any        `json:"quantity,omitempty"`
-	StartTimeWindow         any        `json:"start_time_window,omitempty"`
-	MaxWait                 *int       `json:"max_wait,omitempty"`
-	Duration                *int       `json:"duration,omitempty"`
-	TargetArrivalTime       *time.Time `json:"target_arrival_time,omitempty"`
-	EarlyArrivalTimePenalty *float64   `json:"early_arrival_time_penalty,omitempty"`
-	LateArrivalTimePenalty  *float64   `json:"late_arrival_time_penalty,omitempty"`
-	CompatibilityAttributes *[]string  `json:"compatibility_attributes,omitempty"`
+	// UnplannedPenalty penalty for not planning a stop.
+	UnplannedPenalty *int `json:"unplanned_penalty,omitempty"`
+	// Quantity of the stop.
+	Quantity any `json:"quantity,omitempty"`
+	// StartTimeWindow time window in which the stop can start service.
+	StartTimeWindow any `json:"start_time_window,omitempty"`
+	// MaxWait maximum waiting duration in seconds at the stop.
+	MaxWait *int `json:"max_wait,omitempty"`
+	// Duration in seconds that the stop takes.
+	Duration *int `json:"duration,omitempty"`
+	// TargetArrivalTime at the stop.
+	TargetArrivalTime *time.Time `json:"target_arrival_time,omitempty"`
+	// EarlyArrivalTimePenalty penalty per second for arriving at the stop before the target arrival time.
+	EarlyArrivalTimePenalty *float64 `json:"early_arrival_time_penalty,omitempty"`
+	// LateArrivalTimePenalty penalty per second for arriving at the stop after the target arrival time.
+	LateArrivalTimePenalty *float64 `json:"late_arrival_time_penalty,omitempty"`
+	// CompatibilityAttributes attributes that the stop is compatible with.
+	CompatibilityAttributes *[]string `json:"compatibility_attributes,omitempty"`
 }
 
 // Vehicle represents a vehicle.
 type Vehicle struct {
```

Comment:
I would copy the (extended) comments of the defaults to the vehicle itself (where applicable).